src/models/layout/step.rs

```diff
@@ -0,0 +1,174 @@
+//! in-toto layout's Step
+
+use std::str::FromStr;
+
+use serde::de::{Deserialize, Deserializer, Error as DeserializeError};
+use serde::ser::{Serialize, Serializer};
+use serde_derive::{Deserialize, Serialize};
+
+use crate::crypto::KeyId;
+use crate::{Error, Result};
+
+/// TODO ArtifactRule
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
+pub struct ArtifactRule {}
+
+/// SupplyChainItem summarizes common fields of the two available supply chain
+/// item types in Inspection and Step.
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
+pub struct SupplyChainItem {
```

Comment:
Hmm, interesting design choice. Part of me is now stuck thinking that we could also get away with doing this as a trait. What's your take on this?

Reply:
Do you mean `SupplyChainItem` as a trait? I think the problem here is that rust doesn't have Anonymous Fields as golang. It can be handled by `#[serde(flatten)]` when a `SupplyChainItem` is being serialized

Reply:
i.e., so probably no? let's go ahead with this then.